name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TastytradePDFExtractor.java

```diff
@@ -0,0 +1,529 @@
+import name.abuchen.portfolio.model.PortfolioTransaction;
+import name.abuchen.portfolio.money.Money;
+import name.abuchen.portfolio.money.Values;
+import name.abuchen.portfolio.util.OccOsiSymbology;
+
+@SuppressWarnings("nls")
+public class TastytradePDFExtractor extends AbstractPDFExtractor
+{
+    private List<BuySellEntry> m_optionTradesInImport = new ArrayList<>();
+
+    public TastytradePDFExtractor(Client client)
+    {
+        super(client);
+        addBankIdentifier("TASTYWORKS");
+        addBankIdentifier("tastytrade");
+        addBankIdentifier("Apex Clearing Corporation");
+
+        addSummaryStatementBuySellTransaction(); // Purchase and sale ( multiple settlements )
+        addDepotStatementTransaction(); // Securities account transactions ( Settlement account )
+        addDividendTransaction(); // Dividends
+    }
+
+    @Override
+    public String getLabel()
+    {
+        return "tastytrade, Inc.";
+    }
+
+    private void addDividendTransaction()
+    {
+        DocumentType type = new DocumentType("A[\\s]+C[\\s]+C[\\s]+O[\\s]+U[\\s]+N[\\s]+T[\\s]+"
+                        + "S[\\s]+T[\\s]+A[\\s]+T[\\s]+E[\\s]+M[\\s]+E[\\s]+N[\\s]+T");
+        this.addDocumentTyp(type);
+
+        Block firstRelevantLine = new Block("^DIVIDEND[\\s]+[\\d]{2}/[\\d]{2}/[\\d]{2}[\\s]+[A-Z][\\s]+.*$");
+        type.addBlock(firstRelevantLine);
+
+        Transaction<AccountTransaction> pdfTransaction = new Transaction<>();
+        pdfTransaction.subject(() -> {
+            AccountTransaction entry = new AccountTransaction();
+            entry.setType(AccountTransaction.Type.DIVIDENDS);
+            entry.setCurrencyCode("USD");
+            return entry;
+        });
+        firstRelevantLine.set(pdfTransaction);
+
+        pdfTransaction
+            // DIVIDEND 06/28/23 M PROSHARES ULTRAPRO S&P500
+            // 0.145499 29.10
+            .section("date", "name", "price", "amount")
+            .match("DIVIDEND[\\s]+(?<date>[\\d]{2}/[\\d]{2}/[\\d]{2})[\\s]+[A-Z][\\s]+(?<name>[\\/\\&\\s\\w]+)[\\s]+"
+                            + "(\\$)?(?<price>[\\.,\\d]+)[\\s]+(\\$)?(?<amount>[\\.,\\d]+)")
+            .assign((t, v) -> {
+                v.put("currency", "USD");
+                t.setDateTime(asDate(v.get("date")));
+                t.setAmount(asAmount(v.get("amount"), "en", "US"));
+            })
+            .oneOf(
+                // CASH DIV ON 200 SHS WH 8.73
+                section -> section
+                .attributes("tax", "shares")
+                .match("CASH[\\s]+DIV[\\s]+ON[\\s]+(?<shares>[\\.,\\d]+)[\\s]+SHS[\\s]+WH[\\s]+(?<tax>[\\.,\\d]+)")
+                .assign((t, v) -> {
+                    t.setShares(asShares(v.get("shares"), "en", "US"));
+                    v.put("currency", "USD");
+                    Money tax = Money.of("USD", asAmount(v.get("tax"), "en", "US"));
+                    ExtractorUtils.checkAndSetTax(tax, t, type.getCurrentContext());
```

Comment:
```
                ExtractorUtils.checkAndSetTax(tax, t, type.getCurrentContext());
```
Maybe you use this method in the same example... see in the other PDF importer
(addTaxesSectionsTransaction / addFeesSectionsTransaction)

Reply:
Can you elaborate this a bit more?

Reply:
You could simply look at the other PDF importers and adopt their structure. You will find that they are always structured in the same way in order to achieve standardization. :-)